Initial draft of identity docs (#5133)
camunda/camunda-docs: docs/self-managed/identity/orchestration-identity/configuration.md

---
id: configuration
title: "Configuration"
description: "Learn about the Identity configuration options available in your Orchestration cluster."
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

As a Spring Boot application, Identity supports any standard
[Spring configuration](https://docs.spring.io/spring-boot/reference/features/external-config.html) method.

<!-- updates must be made to BOTH tables -->
<Tabs>
  <TabItem value="env" label="Environment variables" default>

> | Environment variable                                  | Description                                                                                       | Default value       |

Reviewer: 💭 These variables are applicable to all components in the orchestration cluster so I think we need to find a central home for these. As per @megglos [suggestion](https://github.com/camunda/documentation-team/issues/465) I wonder if we can create a new area for the Core components? 

It may be that its more important to just get this in place regardless and then adjust as we go, happy to align off PR.

Author: Followed up on the related issue, but I think ideally we move forward with this iteration, and then make additional Core-related changes in subsequent PRs: https://github.com/camunda/documentation-team/issues/465#issuecomment-2701824081

Reviewer 2: In favor of having this as a first increment too.

---
id: configuration
title: "Configuration"
description: "Learn about the Identity configuration options available in your Orchestration cluster."
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

:::note

> As a Spring Boot application, Identity supports any standard

Reviewer 2: 🔧  effectively Identity is part of the Camunda 8 application, I would thus keep this statement general with regards to Camunda 8 as spring boot application.
```suggestion
As a Spring Boot application, Camunda 8 supports any standard
```

[Spring configuration](https://docs.spring.io/spring-boot/reference/features/external-config.html) method.
:::

> The following variables apply globally to all components within the Camunda Orchestration core: Zeebe, Operate, and Tasklist.

Reviewer 2: I think that Glossary needs updating after this decision https://camunda.slack.com/archives/C01H4NG9XDY/p1740070705216559 where from my understanding Optimize is not considered part of the core orchestration cluster but a separate component.

can you confirm @RomanJRW @felix-mueller @romansmirnov ?

> The following variables apply globally to all components within the Camunda Orchestration core: Zeebe, Operate, and Tasklist.

Reviewer 3: Yes, this would be true.

The kick-off for Optimize in a dedicated app happened earlier this week. One of the defined action items was for the DRI (@buccarel) to reach out to the Docs team and specify what docs changes need as a result of this decision. This needs a more holistic approach than just tweaks imo

> The following variables apply globally to all components within the Camunda Orchestration core: Zeebe, Operate, and Tasklist.

Reviewer 2: 🔧  the config applies also to identity itself too,
```suggestion
The following variables apply globally to all components within the Camunda Orchestration core: Zeebe, Operate, Tasklist and Identity.
```

<!-- updates must be made to BOTH tables -->

> <Tabs>

Reviewer 2: ❓ what is the default config we want to refer to here? Because this differs by distribution, whether it's c8run or helm, based on the second tab it seems helm? I can then provide the actual defaults next. Some like `SPRING_PROFILES_ACTIVE` are also not relevant then as the user can't control them.

Reviewer 2: I see in installation we refer to this as reference from a c8run perspective, I woudl thus suggest to have tabs by distribution c8run and helm. I would omit the ENV variable tab and instead extend the spring boot note by the fact that all properties can be provided as environment variables too, like we do in other places of the docs like here https://docs.camunda.io/docs/next/self-managed/zeebe-deployment/configuration/environment-variables/#environment-variables-for-configuration

Author: Ideally I think this guide should reflect Helm (and other production-ready distributions when ready) and not C8Run at all, as C8Run is unique, not for production, and its configuration requirements are largely handled on its own page. I assume that other production-ready distributions, when available, would reflect the same defaults as Helm?

We've historically had difficulty with users converting from Helm config to environment variables, and the docs team has discussed listing them out separately as we can - @akeller do you have any input here? My thinking was to start having this available for Docker and other distributions down the line, to prevent our docs from being as Helm-biased a they are now.

Reviewer 4: My current understanding is C8Run is for onboarding and non-production use cases to get up and running as quickly as possible (and maybe even, ideally, with no configuration?). Since we prioritize the Java Spring dev during onboarding, leading with the Spring configuration makes sense. 

Once we start talking about production distributions, things are a bit unclear to me - what's our happy path or recommendation when it comes to config? Default config feels like a PM decision and I would defer to @FarkasRabai and/or @theburi. 

I do think we need to be super explicit about Helm config vs. environment variables vs. properties and not expect our users to do the mapping themselves and add `_` or `.` appropriately. We can't just say "follow this pattern" and document the configuration fully in only one way. 

🤞 hope this helps. LMK if there if anything else I can clarify.

| ----------------------------------------------------- | ------------------------------------------------------------------------------------------------- | ------------------- |
| `SPRING_PROFILES_ACTIVE`                              | **Note:** This property will be deprecated as additional authentication methods become available. | `consolidated-auth` |
| `CAMUNDA_SECURITY_AUTHENTICATION_METHOD`              | The authentication method to use.                                                                 | `basic`             |
| `CAMUNDA_SECURITY_AUTHENTICATION_UNAUTHENTICATED-API` | If the API is enabled without authentication.                                                     | `true`              |
| `CAMUNDA_PERSISTENT_SESSIONS_ENABLED`                 | Enables shared authentication between the Orchestration web applications (Operate and Tasklist).  | `true`              |
| `CAMUNDA_SECURITY_AUTHORIZATIONS_ENABLED`             | If authorizations are enabled.                                                                    | `true`              |

> | `CAMUNDA_SECURITY_INITIALIZATION_USERS[0]_USERNAME`   | The username of the first user.                                                                   | `demo`              |

Reviewer 4: Is this "first user" persistent after the initial configuration? Should the example really be "admin" or something that suggests elevated privileges? Or is this just a case where we need any kind of user to populate this table and get things moving?

Reviewer: Good points @akeller - the is persisted in the deployment the choice of `demo` was to match the C7 behaviour IIRC where you have a initial user that can do anything without needing to perform additional setup

| `CAMUNDA_SECURITY_INITIALIZATION_USERS[0]_NAME`     | The name of the first user.                                                                       | Demo                          |
| `CAMUNDA_SECURITY_INITIALIZATION_USERS[0]_EMAIL`    | The email address of the first user.                                                              | `demo@demo.com`               |

  </TabItem>
  <TabItem value="helm" label="Helm properties" default>

| Helm property                                       | Description                                                                                       | Default value (Camunda 8 Run) |
| --------------------------------------------------- | ------------------------------------------------------------------------------------------------- | ----------------------------- |
| `spring.profiles.active`                            | **Note:** This property will be deprecated as additional authentication methods become available. | `consolidated-auth`           |
| `camunda.security.authentication.method`            | The authentication method to use.                                                                 | `basic`                       |

> | `camunda.security.authentication.unprotected-api`   | If the API cane be used without authentication.                                                   | `true`                        |

Reviewer: ```suggestion
| `camunda.security.authentication.unprotected-api`   | If the API can be used without authentication.                                                   | `true`                        |
```

| `CAMUNDA_SECURITY_INITIALIZATION_USERS[0]_NAME`     | The name of the first user.                                                                       | Demo                |
| `CAMUNDA_SECURITY_INITIALIZATION_USERS[0]_EMAIL`    | The email address of the first user.                                                              | `demo@demo.com`     |

  </TabItem>
  <TabItem value="helm" label="Helm properties" default>

| Helm property                                       | Description                                                                                       | Default value       |
| --------------------------------------------------- | ------------------------------------------------------------------------------------------------- | ------------------- |
| `spring.profiles.active`                            | **Note:** This property will be deprecated as additional authentication methods become available. | `consolidated-auth` |
| `camunda.security.authentication.method`            | The authentication method to use.                                                                 | `basic`             |

> | `camunda.security.authentication.unprotected-api`   | If the API is enabled without authentication.                                                     | `true`              |

Reviewer: ```suggestion
| `camunda.security.authentication.unprotected-api`   | If the API can be used without authentication.                                                     | `true`              |
```
